content/root/best-practices-guide.textile

```diff
@@ -0,0 +1,89 @@
+
+h4. 5. Choosing between Realtime/ REST libraries and Protocol Adapters 
+
+* Want to maintain a persistent connection with Ably's Realtime platform, so you can subscribe to messages sent on a channel, or be notified whenever the presence set changes? Use the "realtime library":/realtime.
+* Want to just publish data to the platform, don't care about subscribing, and don't want to manage channel attachment? Use the "REST library":/rest.
+* Want to publish at very high volumes or with the lowest possible latency? "The realtime libary can publish at higher volumes (due to message pipelining) and lower latencies than is possible with the REST library":https://support.ably.io/solution/articles/3000060319-should-i-use-the-rest-or-realtime-library-.
+* Want to get one&#45;time info about the presence set of a channel? Use the "REST library":/rest.
+* Want to set up realtime communication with IoT devices on platforms for which there's no native Ably client library available? Use the "MQTT protocol adapter":/mqtt.
+* Want to migrate from PubNub/Pusher? Use the relevant "protocol adapter":/adapters to "ease the pain of migration":https://support.ably.io/solution/articles/3000054127-can-you-help-ease-the-pain-of-migration-from-another-realtime-service-: you can switch to Ably right away while sticking with your current client libraries, and gradually migrate to native Ably clients.
+
+h4. 6. Subscribing to events on the server&#45;side
+
+* Subscribing to events on the server side using the standard Pub/Sub pattern can be a bad idea for a number of reasons as explained in this "blog article":https://blog.ably.io/message-queues-the-right-way-to-process-and-work-with-realtime-data-on-your-servers-2d15985301f8. With Message Queues, you can set up multiple worker servers to distribute the load of the messages received from Ably while ensuring each of the message gets processed exactly once by any one of your worker servers.
+
+h4. 7. WebHooks/Reactor Functions vs. Message Queues &#45; when to use what?
+
+* "WebHooks":/general/webhooks and "Reactor Functions":/general/functions work well for low or medium message volumes, but "Reactor Queues":/general/queues will scale better to high message volumes. For very high volumes, we recommend using "Reactor Firehose":/general/firehose.
+
+h4. 8. Multiplexing channels
+
+* In almost all cases you should always instantiate a single Ably client library instance per web page or server. Don't forget that the Client Library SDK can "multiplex":https://support.ably.io/support/solutions/articles/3000058406-do-you-support-channel-groups-and-multiplexing- many channels over a single connection.
```

Comment:
'In almost all cases you should always...', 'almost' and 'always' contradict, could simply go for 'In most cases you should instantiate...', or if you want emphasis on how often maybe 'You should almost always instantiate...'.